rewrite-java-test/src/test/java/org/openrewrite/java/UsePrecompiledRegExpForReplaceAllTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class UsePrecompiledRegExpForReplaceAllTest implements RewriteTest {
```

Comment:
I think it would be great to also have a test where a replacement occurs in a nested class as well as one for an anonymous subclass. In both cases we should then have a second `replaceAll()` call in the scope of the outer class. I just think it would be good to make sure we don't get any conflicts on the introduced constants in these cases.